pulpcore/app/models/publication.py

```diff
@@ -381,6 +381,9 @@ class Distribution(MasterModel):
         repository_version (models.ForeignKey): RepositoryVersion to be served.
     """
 
+    # If distribution serves publications, set by subclasses for proper handling in content app
+    publication_based = False
```

Comment:
We will want to update the plugin template with this change, and probably the plugin writer guide too.

Reply:
I'm thinking of renaming this because of the behavior expected by pulp_python. pulp_python allows for content to be served from a publication or live from a repository by using the fact that you can currently access content if you know where it is stored in the content app. I'm thinking maybe `protect_unpublished` or `only_serve_published`.

Reply:
Between those two, the latter makes more sense.  But let's lock down the name soon because the RPM plugin will release tomorrow and the file plugin will need a new release

Reply:
with having little context of the python plugin needs these names do not make much sense to me.

Reply:
Basically pulp_python uses this bug of being able to access unpublished content to allow users to serve content whether they have created a publication or not. So in a sense pulp_python fits under this term `publication_based`, but doesn't want to use this fix since it'll break its current features. This fix is suppose to stop access to unpublished content when using the auto-distribute feature, so renaming the variable will better suite its purpose.

Reply:
I have mixed feelings on this approach, On one hand we claim that this is a bug and we're trying to fix this but still want to enable certain plugin/s taking advantage of this gap. How plugins that have distribution only should read/understand this? `protect_unpublished` or `only_serve_published`.
How about `serve_from_publication`?
re: python plugin. I am confused. Python plugin has publication, but it seems to be a nominal one because content can be consumed regardless of it's presence? What is the purpose of having it then?

Reply:
before approving this PR, I'd like to agree in the variable naming :+1:

Reply:
as per discussion with @mdellweg we should probably have it as constant in caps

Reply:
I want to add, that this is supposed to be a Class constant. So we should make it all caps (however we name it).

Reply:
+1 to @ipanova's suggestion of `serve_from_publication`

And +1 to class constant.

>re: python plugin. I am confused. Python plugin has publication, but it seems to be a nominal one because content can be consumed regardless of it's presence? What is the purpose of having it then?

IIRC it's because the metadata is sufficiently cheap to create on-the-fly due to the way it's accessed, correct me if I'm wrong @gerrod3.

Reply:
```suggestion
    PUBLICATION_BASED = False
```
```suggestion
    SERVE_FROM_PUBLICATION = False
```

In the Exporter refactoring PR we came across the question, how we could derive this value from a repository version to determine whether it's OK to export a repo_version directly or you'd need to go through a publication.
Would it make sense to move this variable to the repository?

Reply:
@mdellweg it should be possible, I think `repo.SERVED_FROM_PUBLICATION` would do it.